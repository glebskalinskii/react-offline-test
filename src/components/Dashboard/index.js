import React, { useState } from 'react';
import Chart from '../Chart';

import './styles.css';

const Dashboard = ({ data }) => {

  const [activeIndex, setActiveIndex] = useState(null);

  const COLORS = [
    '#6aedc7',
    '#39c2c9',
    '#ffce00',
    '#ffa71a',
    '#f866b9',
    '#998ce3',
    '#206953',
    '#f9e9c5',
    '#fb8825',
    '#45494E',
  ];
  
  const generationmixWithKeys = data.generationmix
    .map((energyUnit, index) => ({...energyUnit, colorKey: COLORS[index % COLORS.length]}));

  return <div className="power-dashboard">
    <Chart
      data={generationmixWithKeys}
      activeIndex={activeIndex}
      setActiveIndex={setActiveIndex}
    />
  </div>
}

export default Dashboard;
