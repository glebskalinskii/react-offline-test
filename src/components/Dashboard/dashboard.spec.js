import React from 'react';
import { shallow } from 'enzyme';
import Dashboard from './index';
import testData from '../../../example_api_response.json';

describe('Dashboard', () => {
  test('should render and match snapshot', () => {
    const wrapper = shallow(<Dashboard data={testData.data} />);

    expect(wrapper).toMatchSnapshot();
  });
});
