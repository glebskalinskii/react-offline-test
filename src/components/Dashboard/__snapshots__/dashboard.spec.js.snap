// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard should render and match snapshot 1`] = `
<div
  className="power-dashboard"
>
  <Chart
    activeIndex={null}
    data={
      Array [
        Object {
          "colorKey": "#6aedc7",
          "fuel": "biomass",
          "perc": 4.8,
        },
        Object {
          "colorKey": "#39c2c9",
          "fuel": "coal",
          "perc": 2.5,
        },
        Object {
          "colorKey": "#ffce00",
          "fuel": "imports",
          "perc": 8.7,
        },
        Object {
          "colorKey": "#ffa71a",
          "fuel": "gas",
          "perc": 46.5,
        },
        Object {
          "colorKey": "#f866b9",
          "fuel": "nuclear",
          "perc": 16.1,
        },
        Object {
          "colorKey": "#998ce3",
          "fuel": "other",
          "perc": 0.3,
        },
        Object {
          "colorKey": "#206953",
          "fuel": "hydro",
          "perc": 0.9,
        },
        Object {
          "colorKey": "#f9e9c5",
          "fuel": "solar",
          "perc": 14.6,
        },
        Object {
          "colorKey": "#fb8825",
          "fuel": "wind",
          "perc": 5.6,
        },
      ]
    }
    setActiveIndex={[Function]}
  />
  <Legend
    activeIndex={null}
    data={
      Array [
        Object {
          "colorKey": "#6aedc7",
          "fuel": "biomass",
          "perc": 4.8,
        },
        Object {
          "colorKey": "#39c2c9",
          "fuel": "coal",
          "perc": 2.5,
        },
        Object {
          "colorKey": "#ffce00",
          "fuel": "imports",
          "perc": 8.7,
        },
        Object {
          "colorKey": "#ffa71a",
          "fuel": "gas",
          "perc": 46.5,
        },
        Object {
          "colorKey": "#f866b9",
          "fuel": "nuclear",
          "perc": 16.1,
        },
        Object {
          "colorKey": "#998ce3",
          "fuel": "other",
          "perc": 0.3,
        },
        Object {
          "colorKey": "#206953",
          "fuel": "hydro",
          "perc": 0.9,
        },
        Object {
          "colorKey": "#f9e9c5",
          "fuel": "solar",
          "perc": 14.6,
        },
        Object {
          "colorKey": "#fb8825",
          "fuel": "wind",
          "perc": 5.6,
        },
      ]
    }
    setActiveIndex={[Function]}
  />
</div>
`;
